tests/sync/test_syncing.py

```diff
@@ -91,3 +136,151 @@ def test_complex_sync_all(self, tmpdir: py.path.local):
             pathlib.Path(f'{tmpdir}/syncs/sync-2/group4-file1.txt'),
             pathlib.Path(f'{tmpdir}/syncs/sync-2/group4-file2.txt'),
         ]
+
+
+class TestConfigError:
+    def test_valid_configuration(self, tmpdir: py.path.local):
+        file_name = pathlib.Path(tmpdir / 'config.yml')
+        with file_name.open('w') as f:
+            f.write("""
+            root-dir: ./asdf
+            connections:
+              - name: mytest-plugin
+                plugin: smb
+                username: myuser
+              - name: another-plugin
+                plugin: smb
+                username: otheruser
+            subjects:
+              - name: sync-1
+                sources:
+                  - connection: mytest-plugin
+                    remote-dir: Some/Test/Dir1
+                  - connection: another-plugin
+                    remote-dir: Another/Test/Dir1
+              - name: sync-2
+                sources:
+                  - connection: mytest-plugin
+                    remote-dir: Some/Test/Dir2
+                  - connection: another-plugin
+                    remote-dir: Another/Test/Dir2
+            """)
+        assert syncing.config_error(file_name) is None
+
+    def test_configuration_without_a_file(self, tmpdir: py.path.local):
+        file_name = pathlib.Path(tmpdir / 'config.yml')
+        assert syncing.config_error(file_name) == f'Could not find the file {file_name}'
+
+    def test_configuration_with_an_empty_file(self, tmpdir: py.path.local):
+        file_name = pathlib.Path(tmpdir / 'config.yml')
+        with file_name.open('w') as f:
+            f.write("")
```

Comment:
Similar to above, you can do `config_yml.ensure()` ([docs](http://py.readthedocs.io/en/latest/path.html#py._path.local.LocalPath.ensure)) which just creates an empty file if it doesn't exist.